milabench/validation/ensure_rate.py

```diff
@@ -0,0 +1,42 @@
+from collections import defaultdict
+from dataclasses import dataclass, field
+from typing import List
+
+from .validation import ValidationLayer, BenchLogEntry
+
+
+class _Layer(ValidationLayer):
+    """Makes sure the training rate is generated for each benchmarks"""
+
+    def __init__(self, **kwargs) -> None:
+        self.rates = defaultdict(float)
+        self.errors = 0
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args):
+        pass
+
+    def on_event(self, entry: BenchLogEntry):
+        tag = entry.tag
+
+        if tag not in self.rates:
+            self.rates[tag] = 0
```

Comment:
That shouldn't be necessary, given that `self.rates` is a `defaultdict(float)`.

Reply:
It is necessary because if we are not receiving rates then when doing `for tag, rate in self.rates.items():` we will not be iterating through the missing benches that did not execute the `self.rates[tag] += entry.data.get("rate", 0)`.

I removed the defaultdict to prevent confusion in the future

Reply:
Ah, I see.